lib/exq/redis_queue.ex

```diff
@@ -23,10 +32,35 @@ defmodule Exq.RedisQueue do
   end
 
   def enqueue(redis, namespace, queue, worker, args) do
-    {jid, job} = job_json(queue, worker, args)
+    {jid, job_json} = to_job_json(queue, worker, args)
+    enqueue(redis, namespace, queue, job_json)
+    jid
+  end
+  def enqueue(redis, namespace, job_json) do
+    job = Exq.Json.decode!(job_json)
+    enqueue(redis, namespace, job["queue"], job_json)
+    job["jid"]
+  end
+  def enqueue(redis, namespace, queue, job_json) do
     [{:ok, _}, {:ok, _}] = :eredis.qp(redis, [
       ["SADD", full_key(namespace, "queues"), queue],
-      ["RPUSH", queue_key(namespace, queue), job]])
+      ["RPUSH", queue_key(namespace, queue), job_json]])
+  end
+
+  def enqueue_at(redis, namespace, queue, time, worker, args) do
+    enqueued_at = DateFormat.format!(Date.from(time, :timestamp) |> Date.local, "{ISO}")
+    {jid, job_json} = to_job_json(queue, worker, args, enqueued_at)
+    score = time_to_score(time)
+    Exq.Redis.zadd!(redis, scheduled_queue_key(namespace), score, job_json)
+    jid
+  end
+
+  def enqueue_in(redis, namespace, queue, offset, worker, args) do
+    time = Time.add(Time.now, Time.from(offset, :secs))
+    enqueued_at = DateFormat.format!(Date.from(time, :timestamp) |> Date.local, "{ISO}")
```

Comment:
Just noticed this - can probably just call enqueue_at with the specific time instead of redefining it here